spec/features/saml/saml_spec.rb

```diff
@@ -167,6 +167,18 @@
   end
 
   context 'visiting /api/saml/logout' do
+    context 'via SP-initiated logout when not logged in to IdP' do
+      it 'does not raise an exception' do
+        request = OneLogin::RubySaml::Logoutrequest.new
+        settings = sp1_saml_settings
+        settings.name_identifier_value = SecureRandom.uuid
+
+        expect do
+          visit request.create(settings)
+        end.to_not raise_error
+      end
+    end
```

Comment:
I think what @amoose is saying is that the spec that starts on line 263 is for *IdP-initiated* logout, not *SP-initiated* logout. In that test, after the user's session has timed out, they visit `/api/saml/logout` directly, which is an IdP URL.